src/flatpakManifest.ts

```diff
@@ -27,7 +27,26 @@ export class FlatpakManifest {
         this.buildDir = path.join(this.workspace, '.flatpak')
         this.repoDir = path.join(this.buildDir, 'repo')
         this.stateDir = path.join(this.buildDir, 'flatpak-builder')
-        this.stateFile = path.join(this.buildDir, 'pipeline.json')
+    }
+
+    /**
+     * Update the fields of this
+     * @param other the more updated FlatpakManifest to get new info from
+     * @returns whether to request for a rebuild
+     */
+    updateFieldsFrom(other: FlatpakManifest): boolean {
```

Comment:
Do we need to update the fields here? normally we would have separate FlatpakManifest or not?

Reply:
Either could work, but the intent is shown better and clearer this way.

Reply:
I just noticed you are calling this when the manifest receives changes, I think the best way to handle this is to just the new modified manifest as the active one (if it has the same uri as the currently active one)

Reply:
changed to that behavior, and since an `activeManifestChanged` triggers to clear `repoDir`. It will ask to initialize a build every manifest modification. So I guess we can remove that message and just handle the build initialization automatically on a build or rebuild command.

Reply:
> It will ask to initialize a build every manifest modification. 

But normally we already know if the manifest uri changed, if not and that manifest state says it was initialized that should give us enough info to know whether we should show the dialog or not

Reply:
Nvm, that was not the reason. `===` operator in js doesn't check by value, but by instance. So had to use `util.isDeepStrictEqual`,  either way, I think it is still better to remove that message, since running any build command should initialize a build

Reply:
Oops, sorry. I didn't notice your last message. Was it okay though to remove the message with this reason?


> running any build command should initialize a build

Reply:
>  running any build command should initialize a build

It should, but my intention with that dialog is that someone not familiar with Flatpak/VScode could get an idea how to start a build and get the application up and running easily